python/cuml/tests/test_device_selection.py

```diff
@@ -0,0 +1,189 @@
+    sk_model = skLinearRegression(fit_intercept=request.param['fit_intercept'],
+                                  normalize=request.param['normalize'])
+    sk_model.fit(X_train, y_train)
+
+    input_type = request.param['input_type']
+
+    if input_type == 'dataframe':
+        modified_y_train = pd.Series(y_train)
+    elif input_type == 'cudf':
+        modified_y_train = cudf.Series(y_train)
+    else:
+        modified_y_train = to_output_type(y_train, input_type)
+
+    return {
+        'fit_intercept': request.param['fit_intercept'],
+        'normalize': request.param['normalize'],
+        'X_train': to_output_type(X_train, input_type),
+        'y_train': modified_y_train,
+        'X_test': to_output_type(X_test, input_type),
+        'ref_y_test': sk_model.predict(X_test)
+    }
+
+
+def test_train_cpu_infer_cpu(lr_data):
+    model = LinearRegression(fit_intercept=lr_data['fit_intercept'],
+                             normalize=lr_data['normalize'])
+    with using_device_type('cpu'):
+        model.fit(lr_data['X_train'], lr_data['y_train'])
+        cu_pred = model.predict(lr_data['X_test'])
+
+    cu_pred = to_output_type(cu_pred, 'numpy').flatten()
+    np.testing.assert_allclose(lr_data['ref_y_test'], cu_pred)
+
+
+def test_train_gpu_infer_cpu(lr_data):
+    model = LinearRegression(fit_intercept=lr_data['fit_intercept'],
+                             normalize=lr_data['normalize'])
+    with using_device_type('gpu'):
+        model.fit(lr_data['X_train'], lr_data['y_train'])
+    with using_device_type('cpu'):
+        cu_pred = model.predict(lr_data['X_test'])
+
+    cu_pred = to_output_type(cu_pred, 'numpy').flatten()
+    np.testing.assert_allclose(lr_data['ref_y_test'], cu_pred)
+
+
+def test_train_cpu_infer_gpu(lr_data):
+    model = LinearRegression(fit_intercept=lr_data['fit_intercept'],
+                             normalize=lr_data['normalize'])
+    with using_device_type('cpu'):
+        model.fit(lr_data['X_train'], lr_data['y_train'])
+    with using_device_type('gpu'):
+        cu_pred = model.predict(lr_data['X_test'])
+
+    cu_pred = to_output_type(cu_pred, 'numpy').flatten()
+    np.testing.assert_allclose(lr_data['ref_y_test'], cu_pred)
+
+
+def test_train_gpu_infer_gpu(lr_data):
+    model = LinearRegression(fit_intercept=lr_data['fit_intercept'],
+                             normalize=lr_data['normalize'])
+    with using_device_type('gpu'):
+        model.fit(lr_data['X_train'], lr_data['y_train'])
+        cu_pred = model.predict(lr_data['X_test'])
+
+    cu_pred = to_output_type(cu_pred, 'numpy').flatten()
+    np.testing.assert_allclose(lr_data['ref_y_test'], cu_pred)
+    sk_model = skLinearRegression(fit_intercept=lr_data['fit_intercept'],
+                                  normalize=lr_data['normalize'])
+    sk_model.fit(X_train, y_train)
+    sk_pred = sk_model.predict(X_test)
+    np.testing.assert_allclose(sk_pred, cu_pred)
+
+
+@pytest.mark.parametrize('fit_intercept', [False, True])
```

Comment:
Just a note for future models- it seems this test could be made more generic by treating this dict of lists as a kwargs for a given estimator type